app/controllers/application_controller.rb

```diff
@@ -10,6 +10,7 @@ class ApplicationController < ActionController::API
 
   def with_current_request
     ManageIQ::API::Common::Request.with_request(request) do |current|
+      raise Catalog::NotAuthorized, "User not Entitled" unless current.entitlement.hybrid_cloud?
```

Comment:
I also think we should move `current.entitlement.hybrid_cloud?` into a separate method that can evaluate multiple entitlement methods for flexibility as more come in.

Reply:
moved it out into a fancy method that sends an array of methods to the entitlement class and returns whether they're all `true`.